praisonai_tools/n8n/n8n_workflow.py

```
@@ -72,78 +72,104 @@ def run(
+                    activate_response.raise_for_status()
+                    logger.info(f"Activated workflow {workflow_id}")
+                
+                # Construct webhook URL
+                # Use /webhook-test/ for test mode, /webhook/ for production
+                webhook_url = f"{self.n8n_url}/webhook/{path}"
+                return webhook_url
+                
+        except httpx.HTTPStatusError as e:
+            logger.error(f"n8n API error fetching workflow {workflow_id}: {e.response.status_code} - {e.response.text}")
+            return {"error": f"HTTP {e.response.status_code}: {e.response.text}"}
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)